src/test/java/org/junitpioneer/jupiter/resource/ResourcesParallelismTests.java

```diff
@@ -0,0 +1,317 @@
+ */
+
+package org.junitpioneer.jupiter.resource;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTimeoutPreemptively;
+import static org.junit.jupiter.api.parallel.ExecutionMode.SAME_THREAD;
+import static org.junitpioneer.jupiter.resource.Scope.GLOBAL;
+import static org.junitpioneer.testkit.assertion.PioneerAssert.assertThat;
+
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+import org.assertj.core.api.Condition;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestFactory;
+import org.junit.jupiter.api.extension.InvocationInterceptor;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.junitpioneer.testkit.ExecutionResults;
+import org.junitpioneer.testkit.PioneerTestKit;
+
+class ResourcesParallelismTests {
+
+	@DisplayName("when a number of shared resources are used in the same test suite")
+	@Nested
+	class WhenANumberOfSharedResourcesAreUsedInSameTestSuiteTests {
+
+		@DisplayName("then the tests do not run in parallel")
+		@Execution(SAME_THREAD)
+		@Test
+		void thenTestsDoNotRunInParallel() {
+			ExecutionResults executionResults = assertTimeoutPreemptively(Duration.ofSeconds(15),
+				() -> PioneerTestKit.executeTestClass(ThrowIfTestsRunInParallelTestCases.class),
+				"The tests in ThrowIfTestsRunInParallelTestCases became deadlocked!");
+			assertThat(executionResults).hasNumberOfSucceededTests(3);
+		}
+
+		@DisplayName("then the test factories do not run in parallel")
+		@Execution(SAME_THREAD)
+		@Test
+		void thenTestFactoriesDoNotRunInParallel() {
+			ExecutionResults executionResults = assertTimeoutPreemptively(Duration.ofSeconds(15),
+				() -> PioneerTestKit.executeTestClass(ThrowIfTestFactoriesRunInParallelTestCases.class),
+				"The tests in ThrowIfTestFactoriesRunInParallelTestCases became deadlocked!");
+			assertThat(executionResults).hasNumberOfSucceededTests(9);
+		}
+
+		@DisplayName("then the test templates do not run in parallel")
+		@Execution(SAME_THREAD)
+		@Test
+		void thenTestTemplatesDoNotRunInParallel() {
+			ExecutionResults executionResults = assertTimeoutPreemptively(Duration.ofSeconds(15),
+				() -> PioneerTestKit.executeTestClass(ThrowIfTestTemplatesRunInParallelTestCases.class),
+				"The tests in ThrowIfTestTemplatesRunInParallelTestCases became deadlocked!");
+			assertThat(executionResults).hasNumberOfSucceededTests(9);
+		}
+
+		@DisplayName("then the test class constructors do not run in parallel")
+		@Execution(SAME_THREAD)
+		@Test
+		void thenTestClassConstructorsDoNotRunInParallel() {
```

Comment:
Same as above but with "method or constructor".

Reply:
Just like my thinking with https://github.com/junit-pioneer/junit-pioneer/pull/491#discussion_r923871071 and https://github.com/junit-pioneer/junit-pioneer/pull/491#discussion_r923871657, I thought that test constructors could run in parallel if users enabled JUnit 5's parallel mode, and if they were annotated with `@Shared`, then I didn't want anything non-deterministic to happen.

Is my thinking incorrect somewhere again?